File: client2/src/game/components/snare.ts
import { Flag } from './flags'

export class Snare {
  static _id = 'Snare'
  _id = Snare._id

  static _flag = Flag.Snare
  _flag = Flag.Snare

  strength: number

  constructor(strength: number = 0) {
    this.strength = strength
  }
}

File: packages/client2/lib/game/components/snare.dart
import '../../ecs/component.dart';
import '../flags.dart';

class Snare extends Component {
  static const int Flag = ComponentFlags.Snare;
  int flag = Snare.Flag;

  static const String Id = 'Snare';
  String id = Snare.Id;

  String strength;

  Snare({
    this.strength = '',
  });

  @override
  String toString() {
    return 'Snare (strength: $strength)';
  }
}
